### PR TITLE
fix: cluster registration information is overwritten

### PR DESCRIPTION
**Pull request by CeerDecy**
#### What this PR does / why we need it:
fix cluster registration information is overwritten

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=388338&iterationID=12783&type=BUG)

#### Specified Reviewers:

/assign @iutx 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |      fix cluster registration information is overwritten        |
| 🇨🇳 中文    |       修复集群注册信息被覆盖       |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/tools/cluster-manager/dialer/server/tunnel-server.go [hunk 1]
```diff
@@ -74,7 +72,7 @@ func clusterRegister(ctx context.Context, server *remotedialer.Server, rw http.R
 	registerFunc := func(clusterKey string, clusterInfo cluster) {
 		ctx, cancel := context.WithTimeout(context.Background(), registerTimeout)
 		defer cancel()
-		ctx = transport.WithHeader(ctx, metadata.New(map[string]string{httputil.InternalHeader: "cluster-manager"}))
+		ctx = apis.WithInternalClientContext(ctx, "cluster-manager")
```
Comment (iutx) [inline internal/tools/cluster-manager/dialer/server/tunnel-server.go:75]: Use `discover.SvcClusterManager` instead.

**Review by iutx — COMMENTED** [review]
@CeerDecy

**Review by iutx — APPROVED** [review]
LGTM
@sfwn

**Review by iutx — COMMENTED** [review]
@CeerDecy Find a new issue, break merge.

**Comment by CeerDecy** [comment]
> @CeerDecy Find a new issue, break merge.

the issue is not related to current pr

**Review by iutx — APPROVED** [review]
LGTM @sfwn

**Review by sfwn — APPROVED** [review]

**Pull request merged**